File: app/src/main/java/com/bigscreen/mangindo/chapter/ChapterViewHolder.java
package com.bigscreen.mangindo.chapter;


import android.support.v7.widget.RecyclerView;
import android.view.View;

import com.bigscreen.mangindo.databinding.ItemChapterBinding;

import static android.support.v7.widget.RecyclerView.NO_POSITION;

public class ChapterViewHolder extends RecyclerView.ViewHolder {

    private final ItemChapterBinding binding;
    private final OnChapterClickListener clickListener;

    public ChapterViewHolder(ItemChapterBinding binding, OnChapterClickListener clickListener) {
        super(binding.getRoot());
        this.binding = binding;
        this.clickListener = clickListener;
        initViews();
    }

    private void initViews() {
        itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (clickListener != null && getAdapterPosition() != NO_POSITION) {
                    clickListener.onChapterClick(getAdapterPosition());
                }
            }
        });
    }

    public void setViewModel(ChapterItemViewModel viewModel) {
        binding.setViewModel(viewModel);
    }

    public interface OnChapterClickListener {
        void onChapterClick(int position);
    }
}

File: app/src/main/java/com/bigscreen/mangindo/chapter/ChapterViewHolder.kt
package com.bigscreen.mangindo.chapter

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.item_chapter.view.textChapter
import kotlinx.android.synthetic.main.item_chapter.view.textChapterTitle

class ChapterViewHolder(itemView: View,
                        private val clickListener: OnChapterClickListener)
    : RecyclerView.ViewHolder(itemView) {

    init {
        itemView.setOnClickListener {
            if (adapterPosition != RecyclerView.NO_POSITION) {
                clickListener.onChapterClick(adapterPosition)
            }
        }
    }

    fun setViewModel(viewModel: ChapterItemViewModel) {
        itemView.textChapter.text = viewModel.chapterText
        itemView.textChapterTitle.text = viewModel.chapterTitle
    }

    interface OnChapterClickListener {
        fun onChapterClick(position: Int)
    }
}